src/Avalonia.Styling/Controls/NameScope.cs

```diff
@@ -102,15 +71,26 @@ public void Register(string name, object element)
             else
             {
                 _inner.Add(name, element);
-                Registered?.Invoke(this, new NameScopeEventArgs(name, element));
+                if(_pendingSearches.TryGetValue(name, out var tcs))
+                    tcs.SetResult(element);
```

Comment:
We should remove the `tcs` value from the `_pendingSearches` after we set the result.